### PR TITLE
Adding new PROJECT file and CONFIG file

### PR DESCRIPTION
**Pull request by saulmezac**

### CODE REVIEW

**Comment by adam2392** [comment]
See #10 comments by @clairezurn

File: pyautomagic/src/Config.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    NEW_PROJECT = {'LIST_NAME': 'Create New Project', 'NAME': 'Type the name of your new project',
+                   'DATA_FOLDER': 'Choose where your raw data is',
+                   'FOLDER': 'Choose where you want the results to be saved'}
+
+    LOAD_PROJECT = {'List_Name': 'Load and existing project'}
+
+    KEYBOARD_SHORTCUTS = {'GOOD': 'g, 1', 'OK': 'o, 2', 'BAD': 'b , 3', 'INTERPOLATE': 'i, 4', 'NOTRATED': 'n , 5',
+                          'NEXT': 'rightarrow', 'PREVIOUS': 'leftarrow'}
+
+    PREFIX_PATTERN = '^[gobni]+i?p_'
```
Comment (adam2392) [inline pyautomagic/src/Config.py]: note needed.

File: pyautomagic/src/Config.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    RATINGS = {'Good': 'Good', 'Bad': 'Bad', 'Regular': 'Regular', 'Interpolate': 'Interpolate',
+               'NotRated': 'Not Rated'}
+
+    EXTENSIONS = {'mat': '.mat', 'text': '.txt .asc .csv', 'fif': '.fif', 'set': '.set', 'edf': '.edf'}
+
+    PYAUTOMAGIC = 'pyautomagic'
+
+    SRC_FOLDER = 'src'
+
+    GUI_FOLDER = 'gui'
```
Comment (adam2392) [inline pyautomagic/src/Config.py]: not needed?

File: pyautomagic/src/Config.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+    IC_LABEL_CSTS = {'ZIP': 'ICLabel0.3.1.zip', 'RUN_MESSAGE': 'Performing ICLabel (this may take a while...)'}
+
+    EOG_REGRESSION_CSTS = {'RUN_MESSAGE': 'Perform EOG Regression...'}
+
+    GENERAL_CSTS = {'ORIGINAL_FILE': ' ', 'REDUCED_NAME': 'reduced'}
+
+    EEG_SYSTEM_CSTS = {'sys10_20_file': 'standard-10-5-cap385.elp', 'EGI_NAME': 'EGI', 'OTHERS_NAME': 'Others'}
+
+    SETTINGS_PREP = {'pathToSteps': '/allSteps'}
```
Comment (adam2392) [inline pyautomagic/src/Config.py]: are all of these going to be used?

File: tests/pyautomagic/Preprocess_and_Interpolate_Test.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+import os
+import logging
+import ntpath
+import glob
+
+
+logging.basicConfig(format='%(asctime)s - %(levelname)s: %(message)s', level=logging.DEBUG)
+
+
+class Block:
```
Comment (adam2392) [inline tests/pyautomagic/Preprocess_and_Interpolate_Test.py]: Should be named "Test_Block"

File: tests/pyautomagic/test_Project.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+def test_results_folder():
+    test_Project = Project.Project(name, d_folder, ext)
+    assert(test_Project.results_folder == r"C:\Users\saul__000\OneDrive\Escritorio\Johns Hopkins\NeuroData Design I\Project_Test_Folder\derivatives\automagic")
+
+
+def test_file_extension():
+    test_Project = Project.Project(name, d_folder, ext)
+    assert(test_Project.file_extension == ".edf")
+
```
Comment (adam2392) [inline tests/pyautomagic/test_Project.py]: these will all fail as soon as this isn't on your computer anymore

**Comment by saulmezac** [comment]
> Stylistic changes:
> 
> 1. All test classes should be like: "Test_<name_of_stuff>". They should be descriptive of what they are testing.
> 2. All unit test functions should be as above for the corresponding function.
> 3. All integration test functions should be descriptive of what they are testing.
> 
> Code changes:
> 
> 1. Once you have sufficiently tested all the "unit tests", write an integration test that uses all the functions pyautomagic needs from your code, and show that given input "A", you produce output "B". e.g. could be named test_project_automagic_run(), or whatever is descriptive enough
> 
> ^ This is to help you basically. I'm not going to look over a 1k line + .py file, so the better this test is, the easier it is to see what you've done.

I'm having some trouble while testing the results folder without using my local paths. How can I achieve that? The functions that I have to look for the data folder and to set the results folder only works if the specified folder exists in the computer, so how can I test those functions without using my local data folders?

File: tests/pyautomagic/test_Project.py (hunk 5 above)
Comment (saulmezac) [reply tests/pyautomagic/test_Project.py]: > these will all fail as soon as this isn't on your computer anymore

I'm having some trouble while testing the results folder without using my local paths. How can I achieve that? The functions that I have to look for the data folder and to set the results folder only works if the specified folder exists in the computer, so how can I test those functions without using my local data folders?

Comment (saulmezac) [reply tests/pyautomagic/test_Project.py]: I'm having some trouble while testing the results folder without using my local paths. How can I achieve that? The functions that I have to look for the data folder and to set the results folder only works if the specified folder exists in the computer, so how can I test those functions without using my local data folders?

Comment (clairezurn) [reply tests/pyautomagic/test_Project.py]: You can look at my PR for how to make this work.

**Comment by saulmezac** [comment]
Project, Config and test_project (using Claire's dummy test data folder) are updated. Also, minor changes in Block have been done to ensure integration with Project.

**Comment by clairezurn** [comment]
Remove the unused test folder, change the name of your test file to be called test_project.py, and change test_block initialization of the project to match your code, and I think you it looks good to me. You need to get the tests to all pass.

**Comment by saulmezac** [comment]
> Run tests locally and look at the travis build whenever you make changes:
> 
> For example, this is what I got from clicking thru on the travis build:
> 
> ```
> _______________ ERROR collecting tests/pyautomagic/test_block.py _______________
> tests/pyautomagic/test_block.py:28: in <module>
>     dummy_project = Project.Project(
> E   TypeError: __init__() takes 7 positional arguments but 9 were given
> ```
> 
> It's generally good practice to have your tests all pass locally before requesting review. It's hard for me to "review" if the basic stuff is failing.

test_block is not my tests but it's failing on my computer, that's why. It is part of Claire's codes and we already changed the Project initialization on test_block, but we don't know how to change other errors.

**Comment by saulmezac** [comment]
@adam2392 @clairezurn Now all the test pass. We modified test_project, test_block (initialization for Project) and Project preprocess and interpolate is running fine using Block.

**Pull request merged**